### DISCUSSION TITLE
Lazer crashes when changing between apps in Android

### DISCUSSION THREAD

**Discussion by Neon095** [ppy/osu#16512] [Q&A] [answered]
This happens to me now (or in this phone), but before when i changed to another app and then changed back to osu, the game restarts, but never when changing back to the app it remained in the same state as other apps do.
Video:
https://mega.nz/file/7Hx02DgR#CEfBPYltOCQgkVLv8IOEjnYO2ks0Tb2u8dOgOQ59008
[network.log](https://github.com/ppy/osu/files/7891317/network.log)
[performance.log](https://github.com/ppy/osu/files/7891322/performance.log)
[runtime.log](https://github.com/ppy/osu/files/7891324/runtime.log)

**Comment by bdach** [comment]
can you state what android device you're running on?

generally a few releases back changes to support game resume from pause were merged, by preserving the egl context on pause. seems like that has somehow failed on your device, and i half expected this could happen, because android documentation is very weakly worded (saying that the context *might* be preserved on pause), but i haven't seen a report of that until now, so i'm interested to know.

that said there should probably be a fallback path to restart the game from scratch in the old manner rather than outright crash if preserving the context fails.

**Reply by Neon095** [reply]
Im using a redmi note 8 pro with a custom rom (pixel experience) using Android 12, but when i tried the game with another roms, (all of them using Android 11), the game just restarts. But i think is not a problem of the device because i have seen another device too (poco x3 pro) running miui 12.5 android 11 in which the game also restarted. I think maybe it's a problem that appears only from a certain version of android, 11 it could be but I don't know

**Reply by bdach** [reply]
by "the game just restarts", do you mean that it resumes everything as it was, or that it restarts from the intro?

**Reply by bdach** [reply]
also for what it's worth i own an android device with android 11 (oneplus 8t) and resume works perfectly fine here. so i'd say this may be device-specific, or to be more exact, specific to software from a given manufacturer.

**Reply by Neon095** [reply]
With the game just restarts i refer that it restarts from the intro

**Reply by bdach** [reply]
and you're running the same release on all devices? the same version? i'm not sure why the game would restart from scratch only on some devices, sounds like the OS may be stopping the activity on its own entirely rather than just pausing it.

**Reply by Neon095** [reply]
No i wasn't running the same version but i will try it now

**Reply by Neon095** [reply]
Also, here is an android bug report if is it useful or something but idk
[bugreport-aosp_begonia-SQ1D.211205.016.A1-2022-01-18-19-22-37-dumpstate_log-9948.txt](https://github.com/ppy/osu/files/7891520/bugreport-aosp_begonia-SQ1D.211205.016.A1-2022-01-18-19-22-37-dumpstate_log-9948.txt)

**Reply by bdach** [reply]
thanks, the original log excerpt is enough. the cause is pretty clear cut at this point. it's going to be more difficult for me to test potential fixes but i'll see what i can do.

**Reply by Neon095** [reply]
Ok in poco x3 pro with last relase it doesn't actually happens anything it just resumes

**Reply by Neon095** [reply]
Np :)

**Reply by Neon095** [reply]
I tried in sm-t580 running a custom rom with android 12 and it also crashes like in Redmi note 8 pro
Here the logs:
[network.log](https://github.com/ppy/osu/files/7891668/network.log)
[performance.log](https://github.com/ppy/osu/files/7891669/performance.log)
[runtime.log](https://github.com/ppy/osu/files/7891670/runtime.log)

**Reply by Neon095** [reply]
Also, in poco x3 pro at least a few builds ago you can change between apps if you expend in the other apps a little bit of time, but if you spent some time using other apps and then you went back to osu, it closed or restarted (I don't remember exactly what happened) I can't test it right now because I don't have that device but maybe tomorrow i can try it idk

**Reply by bdach** [reply]
re: sm-t580 - yeah that's the same issue.

re: poco x3 closing/restarting after a long time - if it restarted, then there's not much we can do there i'm afraid. android controls the lifetime of all apps and is well within its right to kill one if it's been unused for long enough. if it crashes when trying to resume/restart, then that's worse, but that's still 99% going to be the same issue as the other devices and will be fixed in the same manner.

**Comment by Theighlin** [comment]
I've also been experiencing this. My device is a realme x50 pro running an unmodified version Android 11 realme UI 2.0

**Comment by bdach** [comment, marked as answer]
tracking at https://github.com/ppy/osu-framework/issues/4994
